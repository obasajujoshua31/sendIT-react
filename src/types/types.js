export const signUpUser = "SIGN_UP_USER";
export const signInUser = "SIGN_IN_USER";
export const createOrder = "CREATE_ORDER";
export const createOrderSuccess = "CREATE_ORDER_SUCCESS";
export const createOrderFailure = "CREATE_ORDER_FAILURE";
export const viewOrders = "VIEW_ORDERS";
export const viewOrder = "VIEW_ORDER";
export const updateParcel = "UPDATE_PARCELS";
export const updateLocation = "UPDATE_PARCEL_LOCATION";
export const cancelOrder = "CANCEL_ORDER";
export const updateParcelStatus = "UPDATE_PARCEL_STATUS";
export const signInSuccess = "SIGN_IN_SUCCESS";
export const signInFailure = "SIGN_IN_FAILURE";
export const signUpSuccess = "SIGN_UP_SUCCESS";
export const signUpFailure = "SIGN_UP_FAILURE";
export const isLoggedOut = "LOGGED_OUT";
export const isAdmin = "IS_ADMIN";
