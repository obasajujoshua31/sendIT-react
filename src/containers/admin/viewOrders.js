import React, { Component } from "react";
import loadUserParcels from "../../actions/parcels/admin/getParcels";
import { connect } from "react-redux";
import { withRouter } from "react-router-dom";
import Spinner from "react-md-spinner";
import {
  signInFailure,
  signUpFailure,
  signInSuccess,
  signUpSuccess
} from "../../types/types";
import Order from "../dashboard/order";
import EditModal from "../../components/modals/edit";
import ReactModal from "react-modal";

export class AdminViewOrders extends Component {
  constructor(props) {
    super(props);
    this.state = {
      parcels: [],
      showModal: false,
      parcelId: "",
      isLoading: true
    };

    this.onHandleViewOrder = this.onHandleViewOrder.bind(this);
    this.closeModal = this.closeModal.bind(this);
  }

  componentWillMount() {
    if (process.env.NODE_ENV !== "test") {
      ReactModal.setAppElement("#app");
      ReactModal.defaultStyles.overlay.backgroundColor = "cornsilk";
    }
  }

  onHandleViewOrder(parcelId) {
    this.setState({
      showModal: true,
      parcelId
    });
  }

  closeModal() {
    this.setState({ showModal: false });
  }
  componentDidMount() {
    this.props.loadParcels();
  }

  componentDidUpdate(prevProps) {
    if (prevProps !== this.props) {
      if (this.props.userParcels) {
        this.setState({ isLoading: false });
        this.setState({ parcels: this.props.userParcels });
      }
      if (
        this.props.authStatus === signInFailure ||
        this.props.authStatus === signUpFailure
      ) {
        this.props.history.push("/login");
      }
    }
  }

  render() {
    if (this.state.isLoading) {
      return (
        <div className="container">
          <div className="table-container">
            <div className="container-not-found">
              <Spinner />
            </div>
          </div>
        </div>
      );
    }

    return (
      <React.Fragment>
        <div className="view-order-container">
          <h1>All Available Orders</h1>
          <table className="table" align="center">
            <thead>
              <tr>
                <td>Parcel Id</td>
                <td>Parcel Name</td>
                <td>Pick Up Location</td>
                <td>Destination</td>
                <td>Sent On</td>
                <td>Status</td>
              </tr>
            </thead>
            <tbody>
              {this.state.parcels.map(value => {
                return (
                  <React.Fragment key={value.parcel_id}>
                    <Order
                      key={value.parcel_id}
                      parcel={value}
                      onHandleViewOrder={this.onHandleViewOrder}
                    />
                  </React.Fragment>
                );
              })}
            </tbody>
          </table>
        </div>
        {this.state.showModal && (
          <EditModal
            show={this.state.showModal}
            handleCloseModal={this.closeModal}
            parcelId={this.state.parcelId}
            role={true}
          />
        )}
      </React.Fragment>
    );
  }
}

export const mapDispatchToProps = dispatch => ({
  loadParcels: () => dispatch(loadUserParcels())
});

export const mapStateToProps = ({ parcels, users }) => ({
  userParcels: parcels.parcels,
  authStatus: users.singInStatus,
  role: users.role
});
export default connect(
  mapStateToProps,
  mapDispatchToProps
)(withRouter(AdminViewOrders));
