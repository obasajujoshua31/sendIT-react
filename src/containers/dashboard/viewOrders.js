import React, { Component } from "react";
import loadUserParcels from "../../actions/parcels/getParcels";
import { connect } from "react-redux";
import { withRouter, Link } from "react-router-dom";
import Spinner from "react-md-spinner";
import {
  signInFailure,
  signUpFailure,
  signInSuccess,
  signUpSuccess
} from "../../types/types";
import Order from "./order";
import EditModal from "../../components/modals/edit";
import ReactModal from "react-modal";

export class ViewOrders extends Component {
  constructor(props) {
    super(props);
    this.state = {
      parcels: [],
      showModal: false,
      parcelId: "",
      isLoading: true
    };
    this.onHandleViewOrder = this.onHandleViewOrder.bind(this);
    this.closeModal = this.closeModal.bind(this);
  }

  onHandleViewOrder(parcelId) {
    this.setState({
      showModal: true,
      parcelId,
      authStatus: signInSuccess || signUpSuccess
    });
  }
  closeModal() {
    this.setState({ showModal: false });
  }
  componentWillMount() {
    if (process.env.NODE_ENV !== "test") {
      ReactModal.setAppElement("#app");
      ReactModal.defaultStyles.overlay.backgroundColor = "cornsilk";
    }
  }

  componentDidMount() {
    this.props.loadParcels(this.props.history);
  }

  componentDidUpdate(prevProps) {
    if (prevProps !== this.props) {
      if (this.props.userParcels) {
        this.setState({ isLoading: false });
        this.setState({ parcels: this.props.userParcels });
      }
      if (
        this.props.authStatus === signInFailure ||
        this.props.authStatus === signUpFailure
      ) {
        this.props.history.push("/login");
      }
    }
  }

  render() {
    if (this.state.isLoading) {
      return (
        <div className="container">
          <div className="table-container">
            <div className="container-not-found">
              <Spinner />
            </div>
          </div>
        </div>
      );
    }
    if (!this.state.parcels.length) {
      return (
        <div className="container">
          <div className="table-container">
            <div className="container-not-found">
              <h1>You are Welcome to SendIT Courier Services</h1>
              <h1>You don't have any parcels yet</h1>
              <h1>
                <Link to="/create-order">Click here to create a Parcel</Link>
              </h1>
            </div>
          </div>
        </div>
      );
    }

    return (
      <React.Fragment>
        <div className="container">
          <div className="table-container">
            <h1
              style={{
                textAlign: "center",
                paddingTop: "20px",
                color: "#806B75"
              }}
            >
              Your Orders
            </h1>
            <table className="table" align="center">
              <thead className="thead-dark">
                <tr>
                  <th scope="col">Parcel Id</th>
                  <th scope="col">Parcel Name</th>
                  <th scope="col">Pick Up Location</th>
                  <th scope="col">Destination</th>
                  <th scope="col">Sent On</th>
                  <th scope="col">Status</th>
                </tr>
              </thead>
              <tbody>
                {this.state.parcels.map(value => {
                  return (
                    <React.Fragment key={value.parcel_id}>
                      <Order
                        key={value.parcel_id}
                        parcel={value}
                        onHandleViewOrder={this.onHandleViewOrder}
                      />
                    </React.Fragment>
                  );
                })}
              </tbody>
            </table>
          </div>
        </div>
        {this.state.showModal && (
          <EditModal
            show={this.state.showModal}
            handleCloseModal={this.closeModal}
            parcelId={this.state.parcelId}
            role={false}
            header={"Change Destination"}
          />
        )}
      </React.Fragment>
    );
  }
}

export const mapDispatchToProps = dispatch => ({
  loadParcels: () => dispatch(loadUserParcels())
});

export const mapStateToProps = ({ parcels, users }) => ({
  userParcels: parcels.parcels,
  authStatus: users.singInStatus,
  role: users.role,
});
export default connect(
  mapStateToProps,
  mapDispatchToProps
)(withRouter(ViewOrders));
