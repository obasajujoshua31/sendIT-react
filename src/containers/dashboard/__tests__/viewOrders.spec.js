import React from "react";
import ReactModal from "react-modal";
import { ViewOrders, mapDispatchToProps, mapStateToProps } from "../viewOrders";
import { shallow } from "enzyme";
import sinon from "sinon";
import { signInSuccess, signInFailure } from "../../../types/types";

const props = {
  userParcels: [],
  loadParcels: jest.fn(),
  history: {
    push: jest.fn()
  },
  authStatus: "",
  role: false
};


  const mockParcels = [
    {
      parcel: {
        id: "1",
        status: "PLACED"
      }
    },
    {
      parcel: {
        id: "2",
        status: "DELIVERED"
      }
    },
    {
      parcel: {
        id: "3",
        status: "TRANSITING"
      }
    },
    {
      parcel: {
        id: "4",
        status: "CANCELLED"
      }
    }
  ]

const mockStore = {
  users: {
    singInStatus: signInSuccess
  },
  parcels: {
    parcels: ["Beans"]
  },
  role: false
};
const wrapper = shallow(<ViewOrders {...props} />);
describe("Test for User ViewOrders", () => {
  it("should render ViewOrders component", () => {
    wrapper.setState({
      isLoading: false,
      showModal: false
    });
    expect(wrapper).toMatchSnapshot();
  });

  it("should dispatch loadparcels", () => {
    const dispatchSpy = sinon.spy();
    const dispatchResult = mapDispatchToProps(dispatchSpy);
    dispatchResult.loadParcels();
    expect(dispatchSpy.callCount).toEqual(1);
  });

  it("should return new object for mapstate to props", () => {
    const stateToPropsResult = mapStateToProps(mockStore);
    expect(stateToPropsResult.userParcels).toEqual(["Beans"]);
    expect(stateToPropsResult.authStatus).toEqual(signInSuccess);
  });

  it("should render view orders", () => {
    const viewOrderWrapper = shallow(<ViewOrders {...props} />);
    viewOrderWrapper.setProps({
      userParcels: mockParcels
    });
    expect(viewOrderWrapper).toMatchSnapshot();
  });

  it("should call history.push", () => {
    const viewOrderWrapper = shallow(<ViewOrders {...props} />);
    viewOrderWrapper.setProps({
      authStatus: signInFailure
    });
    expect(viewOrderWrapper.instance().props.history.push).toBeCalled();
  });

  it("should call render parcels", () => {
    const viewOrderWrapper = shallow(<ViewOrders {...props} />);
    viewOrderWrapper.setState({
      parcels: mockParcels,
      showModal: false,
      isLoading: false
    });
    expect(viewOrderWrapper).toMatchSnapshot();
  });

  it("state modal should be false", () => {
    const viewOrderWrapper = shallow(<ViewOrders {...props} />);
    const viewSpy = jest.spyOn(
      viewOrderWrapper.instance(),
      "onHandleViewOrder"
    );
    viewOrderWrapper.setState({
      parcels: mockParcels
    });
    viewSpy("1");
    expect(viewOrderWrapper.instance().state.showModal).toEqual(true);
  });

  it("close Modal should set state to false", () => {
    const viewOrderWrapper = shallow(<ViewOrders {...props} />);
    const viewSpy = jest.spyOn(viewOrderWrapper.instance(), "closeModal");
    viewOrderWrapper.setState({
      parcels: mockParcels
    });
    viewSpy();
    expect(viewOrderWrapper.instance().state.showModal).toEqual(false);
  });
});
