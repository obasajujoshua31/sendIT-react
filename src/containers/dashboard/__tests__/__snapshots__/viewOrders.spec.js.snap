// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test for User ViewOrders should call render parcels 1`] = `
<Fragment>
  <div
    className="container"
  >
    <div
      className="table-container"
    >
      <h1
        style={
          Object {
            "color": "#806B75",
            "paddingTop": "20px",
            "textAlign": "center",
          }
        }
      >
        Your Orders
      </h1>
      <table
        align="center"
        className="table"
      >
        <thead
          className="thead-dark"
        >
          <tr>
            <th
              scope="col"
            >
              Parcel Id
            </th>
            <th
              scope="col"
            >
              Parcel Name
            </th>
            <th
              scope="col"
            >
              Pick Up Location
            </th>
            <th
              scope="col"
            >
              Destination
            </th>
            <th
              scope="col"
            >
              Sent On
            </th>
            <th
              scope="col"
            >
              Status
            </th>
          </tr>
        </thead>
        <tbody>
          <withRouter(Order)
            onHandleViewOrder={[Function]}
            parcel={
              Object {
                "parcel": Object {
                  "id": "1",
                  "status": "PLACED",
                },
              }
            }
          />
          <withRouter(Order)
            onHandleViewOrder={[Function]}
            parcel={
              Object {
                "parcel": Object {
                  "id": "2",
                  "status": "DELIVERED",
                },
              }
            }
          />
          <withRouter(Order)
            onHandleViewOrder={[Function]}
            parcel={
              Object {
                "parcel": Object {
                  "id": "3",
                  "status": "TRANSITING",
                },
              }
            }
          />
          <withRouter(Order)
            onHandleViewOrder={[Function]}
            parcel={
              Object {
                "parcel": Object {
                  "id": "4",
                  "status": "CANCELLED",
                },
              }
            }
          />
        </tbody>
      </table>
    </div>
  </div>
</Fragment>
`;

exports[`Test for User ViewOrders should render ViewOrders component 1`] = `
<div
  className="container"
>
  <div
    className="table-container"
  >
    <div
      className="container-not-found"
    >
      <h1>
        You are Welcome to SendIT Courier Services
      </h1>
      <h1>
        You don't have any parcels yet
      </h1>
      <h1>
        <Link
          replace={false}
          to="/create-order"
        >
          Click here to create a Parcel
        </Link>
      </h1>
    </div>
  </div>
</div>
`;

exports[`Test for User ViewOrders should render view orders 1`] = `
<div
  className="container"
>
  <div
    className="table-container"
  >
    <div
      className="container-not-found"
    >
      <h1>
        You are Welcome to SendIT Courier Services
      </h1>
      <h1>
        You don't have any parcels yet
      </h1>
      <h1>
        <Link
          replace={false}
          to="/create-order"
        >
          Click here to create a Parcel
        </Link>
      </h1>
    </div>
  </div>
</div>
`;
